File: projekat/src/app/entities/rezervacijaDestinacije/rezervacijaDestinacije.ts
import { NumberValueAccessor } from '@angular/forms';
import { Destinacija } from '../destinacija/destinacija';

export class RezervacijaDestinacije {
    id: number;
    idAirCompany: number;
    idDestinacije: number;
    idKlijenta: string;
    cena: number;
    pocetniDatum: string;
    krajnjiDatum:string;
    destinacija:Destinacija;
    zavrseno:boolean;

    constructor(id:number, idAirCompany: number, idDestinacije: number, idKlijenta: string, cena: number,pocetniDatum:string,krajnjiDatum:string,destinacija:Destinacija) {
        this.id = id;
        this.idAirCompany = idAirCompany;
        this.idDestinacije = idDestinacije;
        this.idKlijenta = idKlijenta;
        this.cena = cena;
        this.pocetniDatum = pocetniDatum;
        this.krajnjiDatum = krajnjiDatum;
        this.destinacija=destinacija;
    }

}

File: WebApplication1/WebApplication1/Models/RezervacijaDestinacije.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace WebApplication1.Models
{
    public class RezervacijaDestinacije
    {
        [Key]
        public int Id { get; set; }

        public int IdAirCompany { get; set; }

        public int IdDestinacije { get; set; }

        public double Cena { get; set; }

        public bool Zavrseno { get; set; }

        public string IdKlijenta { get; set; }

        public DateTime PocetniDatum { get; set; }

        public DateTime KrajnjiDatum { get; set; }

        public int OcenaZaKomapaniju { get; set; }
        public int OcenaZaDestinaciju { get; set; }

        [NotMapped]
        public List<DateTime> ZauzetiDatumi { get; set; }

        public string ZauzetiDatumiString { get; set; }

        [NotMapped]
        public Destinacija Destinacija { get; set; }


        public void PretvoriUJson()
        {
            string output = JsonConvert.SerializeObject(ZauzetiDatumi);
            ZauzetiDatumiString = output;
        }

        public void PretvoriUListu()
        {
            if (ZauzetiDatumiString != null)
            {
                List<DateTime> deserializedProduct = JsonConvert.DeserializeObject<List<DateTime>>(ZauzetiDatumiString);
                ZauzetiDatumi = deserializedProduct;
            }
        }
    }
}
